fix: update button font size and padding to prevent overlap

hxrshxz opened the pull request:
## Description
Fixed overlapping text and UI elements on p5.js mobile view (iPhone SE )

Fixes #903 

Screenshots (if applicable)
| Before | After |
|--------|-------|
| <img width="491" height="405" src="https://github.com/user-attachments/assets/13919146-25d8-4eec-8588-cce9aeb1aeca" /> <br> <img width="490" height="603" src="https://github.com/user-attachments/assets/1d374383-931f-434c-b542-29554f5b5441" /> | <img width="485" height="649" src="https://github.com/user-attachments/assets/8d61817d-2b40-4136-a96b-049211b52c06" /> <br> <img width="485" height="649" src="https://github.com/user-attachments/assets/3521176e-9e85-48b1-8866-644a98811dc3" /> |

hxrshxz committed:
fix: update button font size and padding to prevent overlap

hxrshxz committed:
fix: add comments for better understanding of padding and font size adjustments

hxrshxz committed:
Merge branch 'main' into fix-overlap

hxrshxz commented:
@coseeian  please take a look

hxrshxz committed:
Merge branch 'main' into fix-overlap

coseeian commented on src/components/Dropdown/styles.module.scss:
```diff
@@ -113,7 +113,15 @@ button.chevron {
   display: inline-block;
   width: 100%;
+  // 0.75rem font-size for mobile to fit content in smaller screens
+  font-size: 0.75rem;
```
I have some concerns regarding the approach of reducing the font size to fit the text within the container. While it solves the immediate constraint, it introduces a couple of problems:

* Visual Consistency: The smaller text negatively impacts the visual aesthetics of the dropdowns. The text no longer appears to be horizontally centered with the adjacent icons. And it leads to an unbalanced look between the text and icon sizes.

* Readability/Accessibility: I think generally it's better to prioritize larger font sizes for better readability and visibility. Shrinking the text should generally be avoided solely to fix container constraints.

I suggest to explore adjusting the grid column width to leave more space for the dropdowns instead, as long as it aligns with the site's design style guide (if any). I recommend confirming this with the site maintainers.

coseeian commented on styles/global.scss:
```diff
@@ -382,8 +385,7 @@ pre.code-box,
       // 43px is the height of the top mobile bar menu
-      height: calc(50vh - var(--spacing-5xl) - 43px);
-      max-height: calc(50vh - var(--spacing-5xl) - 43px);
+      min-height: calc(50vh - var(--spacing-5xl) - 43px);
```
Curious if there is any concern to remove the height/max-height/min-height CSS in this component?

coseeian reviewed (commented):
I built the PR locally and tested it using the iPhone SE simulator. The issue mentioned in https://github.com/processing/p5.js-website/issues/903 appears to be resolved.

Moving forward, I think it would be valuable to get some input from the site maintainers to ensure this approach aligns with long-term design patterns and accessibility goals.

hxrshxz committed:
Merge branch 'main' into fix-overlap

hxrshxz committed:
Merge upstream/main and address review feedback: revert font size, adjust grid layout

hxrshxz committed:
Fix overlap: Stack settings on mobile and adjust header padding

hxrshxz commented:
**final version** 
<img width="666" height="727" alt="image" src="https://github.com/user-attachments/assets/4497f552-25fd-4d21-b22a-82b0c27a28a6" />

 I reverted the font size change in the dropdowns. Instead, I adjusted the grid layout in the Settings component to allow the dropdowns to stack (full width) on mobile devices. This provides enough space for the content without compromising readability. Regarding the global SCSS, removing the fixed height/max-height was imp for this taller stacked layout on mobile, makeing sure  the container grows dynamically to fit the content."

ksen0 commented:
Thanks @hxrshxz and @coseeian ! Design-wise, I believe this is an improvement.
